common/app/targeting/CampaignAgent.scala

```diff
@@ -5,21 +5,28 @@ import com.gu.targeting.client.CampaignCache
 import conf.Configuration
 import scala.util.control.NonFatal
 import scala.concurrent.Future
+import conf.switches.Switches.Targeting
 
 object CampaignAgent extends Logging with ExecutionContexts {
-  private val agent = AkkaAgent[CampaignCache](CampaignCache(List()))
+  private val agent = AkkaAgent[CampaignCache](CampaignCache(Nil, None))
 
   def refresh(): Future[Unit] = {
     Configuration.targeting.campaignsUrl.map(url => {
-      CampaignCache.fetch(url).flatMap(agent.alter).map(_ => ())
+      CampaignCache.fetch(url, limit = 100).flatMap(agent.alter).map(_ => ())
     }).getOrElse(Future.failed(new BadConfigurationException("Campaigns URL not configured")))
   }
 
-  def getCampaignsForTags(tags: Seq[String]) = try {
-    agent().getCampaignsForTags(tags)
-  } catch {
-    case NonFatal(e) =>
-      log.error("Failed to get campaigns for tags.", e)
-      List()
+  def getCampaignsForTags(tags: Seq[String]) = {
+    if (Targeting.isSwitchedOn) {
+      try {
+        agent().getCampaignsForTags(tags)
```

Comment:
It's more a case of defensive coding, right now the rules engine is very simple and I don't think there are any cases were it could throw an `Exception` in particular but in the future this could change. I'd say it's preferable that the agent itself handle the failure case rather than allowing the exception to propagate to the job scheduler which lacks the context to properly deal with it.